docs/design/fabric-v2+/attestation-interface.md

```diff
@@ -0,0 +1,63 @@
+# Attestation Interface
+
+This document defines the interface for producing and verifying an attestation.
+
+
+## Producing an attestation
+
+This interface is only available inside an enclave.
+
+```
+int init_attestation(
+    uint8_t* params,
+    uint32_t params_length);
+
+```
+```
+int get_attestation(
+    uint8_t statement[32],
+    uint8_t* attestation,
+    uint32_t attestation_length);
+```
+
+
+### Details related to EPID-based SGX attestations
+
+FPC can use EPID-based SGX attestations.
+Such attestations have some peculiarities.
+1. They require to be initialized with some external parameters (SPID, signature revocation list). These parameters are provided through `init_attestation`.
+
+2. They are computed indirectly through a different enclave, called Quoting Enclave. For this reason, the implementation of `get_attestation` uses the following edge functions to retrieve the IAS-verifiable attestation.
+```
+void ocall_init_quote(
+    [out, size=target_len] uint8_t *target, uint32_t target_len,
+    [out, size=egid_len] uint8_t *egid, uint32_t egid_len);
+
+void ocall_get_quote(
+    [in, size=spid_len] uint8_t *spid, uint32_t spid_len,
+    [in, size=report_len] uint8_t *report, uint32_t report_len,
+    [out, size=max_quote_len] uint8_t *quote, uint32_t max_quote_len,
+    [out] uint32_t *actual_quote_len);
+```
+
+3. It must to be sent to IAS, which provides a publicly verifiable report. This step thus involves contacting and authenticating with a web service using the ISV's credentials. For this reason, its completion is delegated to a different entity.
+
+
+## Verifying an attestation
+
+This interface is available both inside and outside of an enclave.
+
+```
+int verify_attestation(
```

Comment:
How is the verification result decoded? We should define a set of error codes this method (and also others) can return. At least basic stuff like, VERIFICATION_SUCCEEEEDS, FAILED

Reply:
I have added a comment. Even more basic: return `0` on error, otherwise it is successful.

Reply:
Related more general question as i was thinking of return codes also in context of #434 (but then punted so far as i was not 100% sure).  Is our general strategy for return values in C:
- return 1/non-zero for true and 0 for false for functions evaluating a predictate
- return 0 for success and non-zero for errors for any other functions
Glancing through some code it seems that's more or less what we have done, but just cross-checking....

Reply:
sounds good, even though I will be confused all the time and mix this stuff. maybe better use boolean return value to clearly differentiate between an error/non-error return value vs an evaluating function.

Reply:
changed to boolean -- [the type is supported in C as of C99](https://en.cppreference.com/w/c/types/boolean)